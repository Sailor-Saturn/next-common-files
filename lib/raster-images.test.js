'use strict';

const { withRasterImages } = require('../index.js');

const options = (overrides) => ({
    dev: true,
    isServer: true,
    ...overrides,
});

const createConfig = () => ({
    module: {
        rules: [],
    },
});

it('should disable static images', () => {
    const config = withRasterImages()();

    expect(config.images.disableStaticImages).toBe(true);

    const config2 = withRasterImages()({
        ...createConfig(),
        images: {
            disableStaticImages: false,
        },
    });

    expect(config2.images.disableStaticImages).toBe(true);
});

it('should configure a unique rule correctly', () => {
    const webpackConfig = withRasterImages()().webpack(createConfig(), options());

    expect(webpackConfig.module.rules).toHaveLength(1);
    expect(webpackConfig.module.rules[0]).toMatchSnapshot();
});

it('should configure a unique rule correctly (dev)', () => {
    const webpackConfig = withRasterImages()().webpack(createConfig(), options({ dev: false }));

    expect(webpackConfig.module.rules).toHaveLength(1);
    expect(webpackConfig.module.rules[0]).toMatchSnapshot();
});

it('should call nextConfig webpack if defined', () => {
    const nextConfig = {
        webpack: jest.fn(() => 'foo'),
    };

    const webpackConfig = withRasterImages()(nextConfig).webpack(createConfig(), options());

    expect(nextConfig.webpack).toHaveBeenCalledTimes(1);
    expect(webpackConfig).toBe('foo');
});
