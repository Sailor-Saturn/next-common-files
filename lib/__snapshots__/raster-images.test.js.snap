// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should configure a unique rule correctly (dev) 1`] = `
Object {
  "loader": "<PROJECT_ROOT>/node_modules/url-loader/dist/cjs.js",
  "options": Object {
    "emitFile": false,
    "limit": 0,
    "name": "images/[name].[contenthash:20].[ext]",
    "outputPath": "static/chunks/media",
    "publicPath": "/_next/static/chunks/media",
  },
  "test": /\\\\\\.\\(png\\|jpg\\|jpeg\\|gif\\|webp\\|ico\\)\\$/,
}
`;

exports[`should configure a unique rule correctly 1`] = `
Object {
  "loader": "<PROJECT_ROOT>/node_modules/url-loader/dist/cjs.js",
  "options": Object {
    "emitFile": false,
    "limit": 0,
    "name": "images/[path][name].[ext]",
    "outputPath": "static/chunks/media",
    "publicPath": "/_next/static/chunks/media",
  },
  "test": /\\\\\\.\\(png\\|jpg\\|jpeg\\|gif\\|webp\\|ico\\)\\$/,
}
`;
